### PR TITLE
KAFKA-13345: Use "delete" cleanup policy for windowed stores if duplicates are enabled

### PR DESCRIPTION
**Pull request by mjsax**
Call for review @guozhangwang @cadonna

### CODE REVIEW

**Review by ableegoldman — APPROVED** [review]
Makes sense to me

File: streams/src/test/java/org/apache/kafka/streams/integration/JoinStoreIntegrationTest.java [hunk 1]
```diff
@@ -131,4 +143,50 @@ public void providingAJoinStoreNameShouldNotMakeTheJoinResultQueriable() throws
...
+        final KStream<String, Integer> right = builder.stream(INPUT_TOPIC_RIGHT, Consumed.with(Serdes.String(), Serdes.Integer()));
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        left.join(
+            right,
+            Integer::sum,
+            JoinWindows.of(ofMillis(100)),
+            StreamJoined.with(Serdes.String(), Serdes.Integer(), Serdes.Integer()).withStoreName("join-store"));
+
+        System.out.println(builder.build().describe());
```
Comment (showuon) [inline streams/src/test/java/org/apache/kafka/streams/integration/JoinStoreIntegrationTest.java]: debug line should be removed.

File: streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractStoreBuilder.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public AbstractStoreBuilder(final String name,
     public StoreBuilder<T> withLoggingEnabled(final Map<String, String> config) {
         Objects.requireNonNull(config, "config can't be null");
         enableLogging = true;
-        logConfig = config;
+        logConfig.putAll(config);
```
Comment (showuon) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractStoreBuilder.java:65]: Nice catch

**Review by showuon — COMMENTED** [review]
LGTM! Left some minor comments.

File: streams/src/test/java/org/apache/kafka/streams/integration/JoinStoreIntegrationTest.java (hunk 1 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/integration/JoinStoreIntegrationTest.java]: Good catch!

**Pull request merged**